docs: clarify Hugo linkTitle guidance (#3843)
open-telemetry/semantic-conventions: CONTRIBUTING.md

--->
```

> Set `linkTitle` when adding a new Markdown file. It drives the website navigation, and shows up relative to the parent document, so it's easy to miss until someone points it out in review.

Reviewer: ```suggestion
Set `linkTitle` when adding a new Markdown file. It drives the [website navigation](https://opentelemetry.io/docs/specs/semconv/) and shows up relative to the parent document.
```

Author: Thanks, I applied this wordings.

Keep it short. Something like `Spans` or `Metrics` works better than spelling out the full convention name again.

> Sentence case is the default, but if the name is an established acronym or product name, keep its usual capitalization.

Reviewer 2: @lmolkova @Jaegeonyoon - thanks for the improvement, but there a mixture of concerns (of the front-matter field purpose and style-guide capitalization information). If you do keep the mixed concerns, how about something like this:

```suggestion
`linkTitle` is a short version of the page's title, shown in the website's
[left side navigation][docsy-nav]. When adding a new Markdown file, keep it to
the sub-area name -- for example, `Spans`, `Metrics`, or `Resources` -- instead of
repeating the page's full title. Omit `linkTitle` when it would be identical to
the title.

For title and heading capitalization, follow the
[OpenTelemetry documentation style guide][otel-style-guide].

[docsy-nav]: https://www.docsy.dev/docs/content/navigation/#side-nav
[otel-style-guide]: https://opentelemetry.io/docs/contributing/style-guide/
```

Otherwise, just drop the last sentence.

Author: Thanks! I applied your suggested wording. I agree this keeps the section focused on the `linkTitle` front-matter field and delegates title/heading capitalization to the OTel documentation style guide.

--->
```

Set `linkTitle` when adding a new Markdown file. It drives the website navigation, and shows up relative to the parent document, so it's easy to miss until someone points it out in review.

Keep it short. Something like `Spans` or `Metrics` works better than spelling out the full convention name again.

Sentence case is the default, but if the name is an established acronym or product name, keep its usual capitalization.

> Also, when renaming or moving files under `docs/`, make sure nothing on the website or any relative links still point to the old path.

Reviewer 3: Un-necessary as the link checker will flag this.

```suggestion
```

Author: Thanks, I removed it.
